### PR TITLE
Com 219 create appointment

### PR DESCRIPTION
**Pull request by arielcamargo**

### CODE REVIEW

File: src/main/java/com/compilercharisma/chameleonbusinessstudio/entity/AppointmentEntity.java [hunk 1]
```diff
@@ -20,108 +20,46 @@
...
-    think it's possible to specify in JPA. Regardless, we'll be migrating to
-    Vendia in the near future, so it likely isn't worth investing time into.
-    */
+
+    private Boolean isCanceled = false;
     
+
+    private Integer totalSlots = 1; // default to one person per appointment
+
+    private List<String> participants;
```
Comment (Matt-Crow) [inline src/main/java/com/compilercharisma/chameleonbusinessstudio/entity/AppointmentEntity.java:54]: Since we're migrating from `Set` to `List`, please add a unit test to ensure duplicate participants cannot be added to an appointment, as `Set`s have built-in checking for this, whereas `List`s do not.

File: src/main/java/com/compilercharisma/chameleonbusinessstudio/entity/AppointmentEntity.java [hunk 2]
```diff
@@ -20,108 +20,46 @@
...
+@NoArgsConstructor
+@Builder
 public class AppointmentEntity implements Serializable {
-    @Id
-    @GeneratedValue(strategy = GenerationType.AUTO)
-    @Setter
-    @Getter
-    @Column(name="appt_id")
-    private int id;
+    private  String _id;
```
Comment (Matt-Crow) [inline src/main/java/com/compilercharisma/chameleonbusinessstudio/entity/AppointmentEntity.java:28]: Changing from an `int` to a `String` will break a couple routes, so please make sure those still function.

File: src/main/java/com/compilercharisma/chameleonbusinessstudio/entity/AppointmentEntity.java [hunk 3]
```diff
@@ -20,108 +20,46 @@
...
-        return sb.toString();
-    }
+//    @Setter
+//    @Getter
+//    @Column(nullable=false)
+//    @ElementCollection
+//    @CollectionTable(joinColumns = @JoinColumn(name = "appt_id"))
+//    @JoinColumn(name = "appt_id")
+//    @OnDelete(action = OnDeleteAction.CASCADE)
+//    private Set<String> tags = new HashSet<>();
```
Comment (Matt-Crow) [inline src/main/java/com/compilercharisma/chameleonbusinessstudio/entity/AppointmentEntity.java:63]: Please do not comment out features, as this leads to a cascading effect where we have to remember to re-enable what we've disabled.

File: src/main/java/com/compilercharisma/chameleonbusinessstudio/service/AppointmentService.java [hunk 4]
```diff
@@ -25,177 +27,192 @@
...
+//    }
+//
+//    public boolean isUserRegistered(AppointmentEntity appt, String email){
+//        return appt.getRegisteredUsers().contains(email);
+//    }
+//
+//    public Mono<Appointment> createAppointment(Appointment appointment)
+//    {
+//        return Mono.just(appointment).flatMap(u -> appointmentRepository.createAppointment(appointment));
+//    }
```
Comment (Matt-Crow) [inline src/main/java/com/compilercharisma/chameleonbusinessstudio/service/AppointmentService.java:217]: As per above. Perhaps we should merge this branch into a `dev` branch until we've reimplemented the repository features.

**Review by Matt-Crow — CHANGES_REQUESTED** [review]

File: src/main/java/com/compilercharisma/chameleonbusinessstudio/entity/AppointmentEntity.java (hunk 2 above)
Comment (dvillavicencio) [reply src/main/java/com/compilercharisma/chameleonbusinessstudio/entity/AppointmentEntity.java:28]: The ID in Vendia is a String

Comment (Matt-Crow) [reply src/main/java/com/compilercharisma/chameleonbusinessstudio/entity/AppointmentEntity.java:28]: Yeah, I know, the routing issue is just an unfortunate byproduct of that. To clarify, I'm not saying "change this back to an int", I'm saying "this change changes a few other things"

**Review by dvillavicencio — APPROVED** [review]
